react-carousel-component/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,65 @@
+import { BsChevronLeft } from 'react-icons/bs';
+import { BsChevronRight } from 'react-icons/bs';
+// import { BsCircleFill } from 'react-icons/bs';
+import { BsCircle } from 'react-icons/bs';
+import { useEffect, useState } from 'react';
+
+type Props = {
+  images: string[];
+};
+
+export default function Carousel({ images }: Props) {
+  const [activeIndex, setActiveIndex] = useState(0);
+
+  function handlePrevClick() {
+    setActiveIndex((activeIndex - 1 + images.length) % images.length);
+  }
+
+  function handleNextClick() {
+    setActiveIndex((activeIndex + 1 + images.length) % images.length);
+  }
+
+  useEffect(() => {
+    setInterval(handleNextClick, 3000);
+  });
```

Comment:
There is no dependency array, so this will execute on every re-render, which means you'll soon have numerous intervals running. Leave it going for a while and you'll see.

You also don't clean up your setInterval, so you are leaking memory.